src/extension/cell.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  Disposable,
+  NotebookCell,
+  NotebookCellExecution,
+  NotebookCellOutput,
+  NotebookCellOutputItem,
+  NotebookController,
+  NotebookEdit,
+  workspace,
+  WorkspaceEdit
+} from 'vscode'
+
+import { OutputType } from '../constants'
+import { CellOutputPayload, Serializer } from '../types'
+import { Mutex } from '../utils/sync'
+
+import { getAnnotations, replaceOutput, validateAnnotations } from './utils'
+
+export class NotebookCellManager {
+  private data = new WeakMap<NotebookCell, NotebookCellOutputManager>()
```

Comment:
I would advocate to use actual private fields, e.g.
```suggestion
  #data = new WeakMap<NotebookCell, NotebookCellOutputManager>()
```

unless you need the object for unit testing purposes.